src/main/core/controller.c

```diff
@@ -538,13 +530,21 @@ gboolean controller_managerFinishedCurrentRound(Controller* controller,
     return newStart < newEnd ? TRUE : FALSE;
 }
 
-gdouble controller_getLatency(Controller* controller, Address* srcAddress, Address* dstAddress) {
+gdouble controller_getLatency(Controller* controller, Address* srcAddress, Address* dstAddress,
+                              bool updateMinTimeJump) {
     MAGIC_ASSERT(controller);
+
+    uint64_t latencyNs = routinginfo_getLatencyNs(controller->routingInfo, controller->ipAssignment,
+                                                  htonl(address_toHostIP(srcAddress)),
+                                                  htonl(address_toHostIP(dstAddress)));
+
+    if (controller->dynamicJumpTime && updateMinTimeJump) {
+        _controller_updateMinTimeJumpNs(controller, latencyNs);
```

Comment:
This can be called by multiple worker threads at once I think, so don't we need some sort of lock inside the `_controller_updateMinTimeJumpNs` function so two threads don't simultaneously write the new jump time?

Reply:
Good catch, we didn't have any locking before, but it worked out since the controller only has one manager which did use its own locking.